### PR TITLE
fix(dependency): update cryptography import

### LINKED ISSUE #22743: cryptography >=39 is incompatible with superset
#### How to reproduce the bug

```bash
> python -m venv venv
> pip install -e .
> python -c "from superset import *"
> Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/ddraper/dev/superset/superset/__init__.py", line 21, in <module>
    from superset.app import create_app
  File "/Users/ddraper/dev/superset/superset/app.py", line 23, in <module>
    from superset.initialization import SupersetAppInitializer
  File "/Users/ddraper/dev/superset/superset/initialization/__init__.py", line 32, in <module>
    from superset.extensions import (
  File "/Users/ddraper/dev/superset/superset/extensions/__init__.py", line 32, in <module>
    from superset.utils.async_query_manager import AsyncQueryManager
  File "/Users/ddraper/dev/superset/superset/utils/async_query_manager.py", line 26, in <module>
    from superset.utils.core import get_user_id
  File "/Users/ddraper/dev/superset/superset/utils/core.py", line 79, in <module>
    from cryptography.hazmat.backends.openssl.x509 import _Certificate
ModuleNotFoundError: No module named 'cryptography.hazmat.backends.openssl.x509'
```

### Expected results

No errors

### Actual results

```> Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/ddraper/dev/superset/superset/__init__.py", line 21, in <module>
    from superset.app import create_app
  File "/Users/ddraper/dev/superset/superset/app.py", line 23, in <module>
    from superset.initialization import SupersetAppInitializer
  File "/Users/ddraper/dev/superset/superset/initialization/__init__.py", line 32, in <module>
    from superset.extensions import (
  File "/Users/ddraper/dev/superset/superset/extensions/__init__.py", line 32, in <module>
    from superset.utils.async_query_manager import AsyncQueryManager
  File "/Users/ddraper/dev/superset/superset/utils/async_query_manager.py", line 26, in <module>
    from superset.utils.core import get_user_id
  File "/Users/ddraper/dev/superset/superset/utils/core.py", line 79, in <module>
    from cryptography.hazmat.backends.openssl.x509 import _Certificate
ModuleNotFoundError: No module named 'cryptography.hazmat.backends.openssl.x509'
```

### Environment

(please complete the following information):

- python version: `python --version`

´``❯ python --version
Python 3.11.1
```
### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [X ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [X ] I have reproduced the issue with at least the latest released version of superset.
- [X ] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

### PR DESCRIPTION
**Pull request by Germandrummer92**
### SUMMARY
* fix the cryptography import

### TESTING INSTRUCTIONS
Before:

``` bash
> python -m venv venv
> . venv/bin/activate
> pip install -e .
> python -c "from superset import *"
> Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/ddraper/dev/superset/superset/__init__.py", line 21, in <module>
    from superset.app import create_app
  File "/Users/ddraper/dev/superset/superset/app.py", line 23, in <module>
    from superset.initialization import SupersetAppInitializer
  File "/Users/ddraper/dev/superset/superset/initialization/__init__.py", line 32, in <module>
    from superset.extensions import (
  File "/Users/ddraper/dev/superset/superset/extensions/__init__.py", line 32, in <module>
    from superset.utils.async_query_manager import AsyncQueryManager
  File "/Users/ddraper/dev/superset/superset/utils/async_query_manager.py", line 26, in <module>
    from superset.utils.core import get_user_id
  File "/Users/ddraper/dev/superset/superset/utils/core.py", line 79, in <module>
    from cryptography.hazmat.backends.openssl.x509 import _Certificate
ModuleNotFoundError: No module named 'cryptography.hazmat.backends.openssl.x509'
```

after:
```bash
> python -m venv venv
> . venv/bin/activate
> pip install -e .
> python -c "from superset import *"
> 
```

### ADDITIONAL INFORMATION

- [X] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

closes  https://github.com/apache/superset/issues/22743

### CODE REVIEW

**Comment by EugeneTorap** [comment]
Hi @dpgaspar @villebro! Can you run CI for this PR?
`cryptography==3.4.7` is released: Mar 25, 2021. Isn't it time to upgrade `cryptography` to `39.0.0`?

**Comment by EugeneTorap** [comment]
Hi @Germandrummer92! Pls, fix `Python Misc / pre-commit (3.8)` CI

**Comment by Germandrummer92** [comment]
@EugeneTorap fixed black, i don't think the mypy failures are related?

**Comment by EugeneTorap** [comment]
@villebro Can you run CI again?

File: superset/utils/core.py [hunk 1]
```diff
@@ -1550,7 +1550,7 @@ def override_user(user: Optional[User], force: bool = True) -> Iterator[Any]:
         delattr(g, "user")
 
 
-def parse_ssl_cert(certificate: str) -> _Certificate:
+def parse_ssl_cert(certificate: str) -> Certificate:
```
Comment (dpgaspar) [inline superset/utils/core.py:1553]: nit: can you use `x509.Certificate` instead or remove the x509 import and import `load_pem_x509_certificate`

Comment (Germandrummer92) [reply superset/utils/core.py:1552]: good point, changed to the direct import of `load_pem_x509_certificate`

**Comment by EugeneTorap** [comment]
@dpgaspar @Germandrummer92 Can we upgrade cryptography to 39.0.0?

**Review by dpgaspar — APPROVED** [review]

**Comment by Germandrummer92** [comment]
@EugeneTorap I ran `pip-compile-multi -P cryptography` thats all thats needed right?

**Comment by Germandrummer92** [comment]
nevermind, i have some weird pip-config, fixing that :)

**Comment by EugeneTorap** [comment]
And also set "cryptography>=39.0.0, <40", in setup.py in order to prevent next major version with breaking changes

**Comment by Germandrummer92** [comment]
should be correct now

**Comment by dpgaspar** [comment]
Please check CI errors

**Comment by EugeneTorap** [comment]
@Germandrummer92 @dpgaspar unit tests are failed because `parse_ssl_cert("abc" + ssl_certificate)` in `test_ssl_certificate_parse` don't throw an Exception in a new version of cryptography.
Should we remove this `self.assertRaises()` line in the unit test?

```python
self.assertRaises(CertificateException, parse_ssl_cert, "abc" + ssl_certificate)
```

**Comment by Germandrummer92** [comment]
thanks for looking into it I didn't get to it yet. I removed the line in question now

File: requirements/base.txt [hunk 2]
```diff
@@ -60,7 +60,7 @@ cron-descriptor==1.2.24
     # via apache-superset
 croniter==1.0.15
     # via apache-superset
-cryptography==3.4.7
+cryptography==39.0.0
```
Comment (dpgaspar) [inline requirements/base.txt:63]: strange change, prob joined the major and minor https://pypi.org/project/cryptography/#history

**Review by dpgaspar — APPROVED** [review]
LGTM, is this ready to merge?

**Pull request merged**